.github/workflows/build.yml

```diff
@@ -32,6 +37,20 @@ on:
         required: false
         type: string
         default: ""
+      secrets-environment:
+        description: Environment to use to access secrets
+        required: false
+        type: string
+        default: ""
+        # secrets-yaml-file:
+        #   description: The secrets.yaml file as a base64 encoded string (from a secret)
+        #   required: false
+        #   type: string
+      secrets-yaml-file-path:
+        description: Location of secrets.yaml file
+        required: false
+        type: string
+        default: "secrets.yaml"
```

Comment:
Since we know the path of each yaml file that is going to be compiled, and `secrets.yaml` needs to be in the same directory as that, we can just use `$(dirname ${{ matrix.file }})/secrets.yaml` to stick it in the right place.